### PR TITLE
RWA-680 Create new work_types table

### PR DESCRIPTION
**Pull request by mkayad**
### JIRA link (if applicable) ###

[RWA-680](https://tools.hmcts.net/jira/browse/RWA-680)

### Change description ###

see ticket

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

**Review by iguannago — DISMISSED** [review]

File: src/main/resources/db/migration/V1.0.4__init_work_types.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+DROP TABLE IF EXISTS work_types;
+CREATE TABLE work_types
+(
+    id TEXT,
+    label TEXT,
+    PRIMARY KEY (id)
+);
+
+INSERT INTO work_types(ID, LABEL) VALUES
```
Comment (raghera) [inline src/main/resources/db/migration/V1.0.4__init_work_types.sql:9]: I know the story had something different but can we name the `id` column as `work_type_id` please?

File: src/main/resources/db/migration/V1.0.4__init_work_types.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    ('hearing-work','Hearing work'),
+    ('upper-tribunal','Upper Tribunal'),
+    ('routine-work','Routine work'),
+    ('decision-making-work','Decision-making work'),
+    ('applications','Applications'),
+    ('priority','Priority'),
+    ('access-requests','Access requests'),
+    ('error-management','Error management');
```
Comment (raghera) [inline src/main/resources/db/migration/V1.0.4__init_work_types.sql:17]: Just thinking if these ought to be `camelCase` or `kebab-case` or `snake_case`.  I think it was put into the story without thinking about consistency elsewhere.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.Getter;
+import lombok.ToString;
+
+import java.io.Serializable;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+
+@EqualsAndHashCode
+@Getter
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java:13]: Better to just have the getter methods than use this annotation

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+@ToString
+@Entity(name = "work_types")
+public class WorkType implements Serializable {
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java:16]: I think the naming convention used by @indvamshi for entities was `<EntityName>Resource`.  Better if we follow that e.g.`WorkTypeResource`

File: src/main/resources/db/migration/V1.0.4__init_work_types.sql (hunk 1 above)
Comment (raghera) [reply src/main/resources/db/migration/V1.0.4__init_work_types.sql:9]: Let's have the identifiers as `snake_case` e.g. `hearing_work`.  This follows the DB conventions better.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java (hunk 4 above)
Comment (mkayad) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java:16]: Done

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java (hunk 3 above)
Comment (mkayad) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/WorkType.java:13]: Done

File: src/main/resources/db/migration/V1.0.4__init_work_types.sql (hunk 2 above)
Comment (mkayad) [reply src/main/resources/db/migration/V1.0.4__init_work_types.sql:17]: I made the ids snake case

File: src/main/resources/db/migration/V1.0.4__init_work_types.sql (hunk 1 above)
Comment (mkayad) [reply src/main/resources/db/migration/V1.0.4__init_work_types.sql]: That is done

**Review by iguannago — DISMISSED** [review]

**Pull request closed without merging**